proto/tendermint/abci/types.proto

```diff
@@ -21,21 +21,25 @@ import "gogoproto/gogo.proto";
 
 message Request {
   oneof value {
-    RequestEcho               echo                 = 1;
-    RequestFlush              flush                = 2;
-    RequestInfo               info                 = 3;
-    RequestInitChain          init_chain           = 4;
-    RequestQuery              query                = 5;
-    RequestBeginBlock         begin_block          = 6;
-    RequestCheckTx            check_tx             = 7;
-    RequestDeliverTx          deliver_tx           = 8;
-    RequestEndBlock           end_block            = 9;
-    RequestCommit             commit               = 10;
-    RequestListSnapshots      list_snapshots       = 11;
-    RequestOfferSnapshot      offer_snapshot       = 12;
-    RequestLoadSnapshotChunk  load_snapshot_chunk  = 13;
-    RequestApplySnapshotChunk apply_snapshot_chunk = 14;
+    RequestEcho                echo                  = 1;
+    RequestFlush               flush                 = 2;
+    RequestInfo                info                  = 3;
+    RequestInitChain           init_chain            = 4;
+    RequestQuery               query                 = 5;
+    RequestBeginBlock          begin_block           = 6 [deprecated = true];
+    RequestCheckTx             check_tx              = 7;
+    RequestDeliverTx           deliver_tx            = 8 [deprecated = true];
+    RequestEndBlock            end_block             = 9 [deprecated = true];
+    RequestCommit              commit                = 10;
+    RequestListSnapshots       list_snapshots        = 11;
+    RequestOfferSnapshot       offer_snapshot        = 12;
+    RequestLoadSnapshotChunk   load_snapshot_chunk   = 13;
+    RequestApplySnapshotChunk  apply_snapshot_chunk  = 14;
+    RequestPrepareProposal     prepare_proposal      = 15;
+    RequestProcessProposal     process_proposal      = 16;
+    RequestFinalizeBlock       finalize_block        = 19;
   }
+  reserved 17, 18; // Future vote extensions
```

Comment:
Is there some reason to reserve these now rather than simply adding them to the end later? Mechanically it's fine, but I don't see a clear reason to insist on tag ordering.

Reply:
I structured the two commits in this PR like this:

* The initial commit contains the whole ABCI++ interface (as it appears in the spec)
* The second commit removes what we are likely to _not_ ship in v0.36 (i.e., vote extension related APIs)

The advantage I see in structuring the commits like this is that, by "git revert"ing the second commit, we can easily get vote extensions back whenever we need them (likely, post v0.36).

As a result, the "reserved" was my way of "removing" vote extensions in the second commit, but of course I'm open to alternative ways of doing this.

Reply:
I think it's not necessary to include the `reserved` tags. Maybe we can just add a comment linking to the vote extensions planned for v0.37

Reply:
OK, I'll turn them into pure comments

Reply:
I see, that makes sense. If we are confident the design will not need to change in such a way that affects these proto messages, I think it's fine to leave reservations, but I agree with @cmwaters that a comment is worthwhile.

(That said: I would _also_ keep the `reserved` notation)

```proto
   reserved 17;  // placeholder for BlahBlah1 in v0.37.
   reserved 18;  // placeholder for BlahBlah2 in v0.37.
```